src/lib.rs

```diff
@@ -427,6 +427,30 @@ impl<L: AnalysisLoader> AnalysisHost<L> {
         result
     }
 
+    pub fn borrow_info(&self, span: &Span) -> AResult<BorrowData> {
+        self.with_analysis(|a|
+            a.def_id_for_span(span)
+             .and_then(|id| a.for_each_crate(|c| {
+                trace!("Searching in crate `{}`", c.name);
+                for (_, b) in c.per_fn_borrows.iter() {
+                    // If we find a `BorrowData` where there's a matching scope, then filter out
+                    // out matching items.
+                    if b.scopes.iter().any(|a| a.ref_id == id) {
```

Comment:
I think I have a working implementation with `BTreeMap`. I'll clean it up and update the PR for review after work (~10 hours from now). The implementation is ... interesting? I'll add a lot of comments to explain what's going on.